lib/api/redfish-1.0/systems.js

```diff
@@ -714,25 +715,43 @@ var listSimpleStorage = controller(function(req, res)  {
     var identifier = req.swagger.params.identifier.value;
     var options = redfish.makeOptions(req, res, identifier);
 
-    return wsman.isDellSystem(identifier)
+    return redfish.getVendorNameById(identifier)
     .then(function(result){
-        if(result.isDell){
+        if(result.vendor === 'Dell'){
             return Promise.resolve(dataFactory(identifier, 'hardware'))
             .then(function(hardware) {
                 options.hardware = hardware;
 
-                var controllers = {};
+                var controllers = new Set();
                 _.forEach(hardware.data.storage.controllers, function(ele) {
                     var id = ele.fQDD.replace(/[:.]/g, '_');  // jshint ignore: line
-                    if(!(id in controllers)) {
-                        controllers[id] = [];
-                    }
-                    controllers[id].push(ele);
+                    controllers.add(id);
                 });
+                options.controllers = Array.from(controllers);
 
-                var ids = [];
-                _.forOwn(controllers, function(val, key) { ids.push(key); });
-                options.controllers = ids;
+                return redfish.render('redfish.1.0.0.simplestoragecollection.json',
+                                'SimpleStorageCollection.json#/definitions/SimpleStorageCollection',
+                                options);
+            }).catch(function(error) {
+                return redfish.handleError(error, res);
+            });
+        } else if(result.vendor === 'Cisco') {
+            return Promise.resolve(dataFactory(identifier, 'UCS:board'))
+            .then(function(storage) {
+                var controllers = [];
+                var sasControllers = storage.data.children['storage-SAS-collection'];
+                var flexControllers = storage.data.children['storage-FlexFlash-collection'];
+                var pchControllers = storage.data.children['storage-PCH-collection'];
```

Comment:
We don't know how many kinds of storage collection there are in UCS system. It is not a good idea to use such kind of enumerating methods.